skbio/sequence/_interval.py

```diff
@@ -0,0 +1,133 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+from __future__ import absolute_import, division, print_function
+
+from ._feature import Feature
+from .intersection import Interval, IntervalTree
+import itertools
+from collections import defaultdict
+
+class IntervalMetadataMixin():
+    ''' Store metadata corresponding to Features and Intervals.
+
+    There are two underlying data structures namely
+    1. An IntervalTree (interval_metadata)
+    2. A hash table indexed by a key and value (feature_table)
+
+    Attributes
+    ----------
+    interval_metadata
+    feature_table
+
+    '''
+    def __init__(self, interval_metadata=None, feature_table=None):
+        self._init_(self, interval_metadata=None, feature_table=None)
+
+    def _init_(self, interval_metadata=None, feature_metadata=None):
```

Comment:
:+1: This gives objects implementing the mixin an obvious API to call. This ends up being more flexible than `super` in practice. Would also be good to implement `eq`, `ne`, `copy`, and `deepcopy`. The Sequence object also defers to its mixins for those operations. 

Basically anything that is wrapped in single underscores is part of the non-public (but not private) interface and exists only for dispatch. Everything else follows the standard rules.